src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php

```diff
@@ -42,16 +42,4 @@
      * @throws \LogicException If the controller can't be found
      */
     public function getController(Request $request);
-
-    /**
-     * Returns the arguments to pass to the controller.
-     *
-     * @param Request  $request    A Request instance
-     * @param callable $controller A PHP callable
-     *
-     * @return array An array of arguments to pass to the controller
-     *
-     * @throws \RuntimeException When value for argument given is not provided
-     */
-    public function getArguments(Request $request, $controller);
```

Comment:
I've thought about it, but that would result in being unable to detect whether the `ControllerResolver` uses the `ArgumentResolver` (the BC layer) or a custom implementation. It would also mean that people that *do* want to implement the `ControllerResolver` without extending the symfony provided implementation, would still have to implement the `getArguments` method. 

I rather have the interfaces in the new layout and have the current implementation provide a BC layer.